iam_builder/templates.py

```diff
@@ -6,207 +6,219 @@
-        {
-            "Sid": "AllowWriteAthenaGlue",
-            "Effect": "Allow",
-            "Action": [
-                "athena:DeleteNamedQuery",
-                "glue:BatchCreatePartition",
-                "glue:BatchDeletePartition",
-                "glue:BatchDeleteTable",
-                "glue:CreateDatabase",
-                "glue:CreatePartition",
-                "glue:CreateTable",
-                "glue:DeleteDatabase",
-                "glue:DeletePartition",
-                "glue:DeleteTable",
-                "glue:UpdateDatabase",
-                "glue:UpdatePartition",
-                "glue:UpdateTable",
-                "glue:CreateUserDefinedFunction",
-                "glue:DeleteUserDefinedFunction",
-                "glue:UpdateUserDefinedFunction"
-            ],
-            "Resource": [
-                "*"
-            ]
-        }
-    ],
-    "glue_job": [
-        {
-            "Sid": "GlueJobActions",
-            "Effect": "Allow",
-            "Action": [
-                "glue:BatchStopJobRun",
-                "glue:CreateJob",
-                "glue:DeleteJob",
-                "glue:GetJob",
-                "glue:GetJobs",
-                "glue:GetJobRun",
-                "glue:GetJobRuns",
-                "glue:StartJobRun",
-                "glue:UpdateJob",
-                "glue:ListJobs",
-                "glue:BatchGetJobs",
-                "glue:GetJobBookmark"
-            ],
-            "Resource": [
-                "*"
-            ]
-        },
-        {
-            "Sid": "CanGetLogs",
-            "Effect": "Allow",
-            "Action": [
-                "logs:GetLogEvents",
-                "logs:CreateLogGroup",
-                "logs:CreateLogStream",
-                "logs:PutLogEvents",
-                "logs:DescribeLogStreams"
-            ],
-            "Resource": [
-                "arn:aws:logs:*:*:/aws-glue/*"
-            ]
-        },
-        {
-            "Sid": "CanGetCloudWatchLogs",
-            "Effect": "Allow",
-            "Action": [
-                "cloudwatch:PutMetricData",
-                "cloudwatch:GetMetricData",
-                "cloudwatch:ListDashboards"
-            ],
-            "Resource": [
-                "*"    
-            ]
-        },
-        {
-            "Sid": "CanReadGlueStuff",
-            "Effect": "Allow",
-            "Action": [
-                "s3:GetObject",
-                "s3:PutObject"
-            ],
-            "Resource": [
-                "arn:aws:s3:::aws-glue-*/*",
-                "arn:aws:s3:::*/*aws-glue-*/*",
-                "arn:aws:s3:::aws-glue-*"
-            ]
-        }
-    ],
-    "decrypt_statement": [
-        {
-            "Sid": "allowDecrypt",
-            "Effect": "Allow",
-            "Action": [
-                "kms:Decrypt"
-            ],
-            "Resource": [
-                "arn:aws:kms:::key/*"
-            ]
-        }
+def iam_lookup(athena_dump_bucket):
```

Comment:
As discussed - agree this should be split out into returning each part of the lookup dict (like it does currently):

Maybe something like:

```python
def iam_lookup(name, **kwargs):
```

Where name would be `"athena_read_access"`, `"glue_job"`, etc.
Then you would pass specific variables like `athena_dump_bucket` through the `kwargs`? This would also make it easier to parameterise other lookups in the future (if we need to do so).

Reply:
Some bits of the lookup don't need any arguments (like `decrypt_statement`). Is there any benefit to returning them from a function? 

I was thinking of using the current approach of having these parts in a static dictionary. Then I'd move the Athena reading parts into a function. This matches the current approach where bits of iam policy like `get_read_only_policy()` or `get_secrets()` are functions.